go-controller/pkg/networkmanager/nad_controller.go

```diff
@@ -180,20 +188,75 @@ func (c *nadController) Stop() {
 	c.networkController.Stop()
 }
 
-// RegisterNADHandler adds functions to be executed during NAD delete/update/add calls
-// usage of this function should be restricted to lightweight, non-blocking operations
-func (c *nadController) RegisterNADHandler(handler handlerFunc) error {
+// RegisterNADHandler adds functions to be executed during NAD delete/update/add calls.
```

Comment:
what if a controller wants something only on startup? like a list of NADs?
so if a controller wants to at initial sync time fetch/list all unmarshalled/parsed NADs from NM is that possible?

asking cause in evpn we were discussing about udncontroller needing the information about vlanid stored somewhere so that on startup this can be parsed and synced correctly - and this will be stored on nad.spec.conf but parsing all of the nads on startup can be expensive so if NM already does that wondering if there is potential to expand the framework to not register on all events, but have a one time list thing exposed? (doesn't have to be part of this PR - just wanted to explore the possibility of if that's doable)